scripts/getPVAliases.py

```diff
@@ -0,0 +1,508 @@
+    """
+    Searches file for regex match and appends result to list
+
+    Parameters
+    ----------
+    file : str
+        The file to read and search. Encoding must be utf-8
+    output : list, optional
+        A list to appead your results to. The default is None.
+    patt : str, optional
+        The regex pattern to search for. The default is None.
+    prefix : str, optional
+        A str prefix to add to each line. The default is ''.
+    color_wrap : Fore, optional
+        Color wrapping using Colorama.Fore. The default is None.
+
+    Returns
+    -------
+    list
+        A list of the search results with the prefix prepended.
+    """
+    if output is None:
+        output = []
+    color = ''
+    reset = ''
+    if color_wrap is not None:
+        color = color_wrap
+        reset = Style.RESET_ALL
+    if os.path.isfile(file) is False:
+        print(f'{file} does not exist')
+        return ''
+    with open(file,'r',encoding='utf-8') as _f:
+        for line in _f.readlines():
+            if re.search(patt,line):
+                output.append(re.sub(patt,color + r'\g<0>' + reset,line))
+        return prefix + prefix.join(output)
+
+
+def clean_ansi(text: str = None):
+    """
+    Removes ANSI escape sequences from a str, including fg/bg formatting.
+    """
+    ansi_escape = re.compile(r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])')
+    return ansi_escape.sub('',text)
+
+
+def fix_json(raw_data: str):
+    """
+    Fixes JSON format of find_ioc output.
+    """
+    # clean empty rows and white space
+    _temp = raw_data.replace(' ','').strip()
+    # capture and fix the keys not properly formatted to str
+    _temp = re.sub(r"(?<!'|\w)\w+(?!')(?=:\s?)",r"'\g<0>'",raw_data)
+    # capture boolean tokens and fix them for json format
+    _temp = re.sub("True","true",_temp)
+    _temp = re.sub("False","false",_temp)
+    # then capture and fix digits not formatted to str
+    _temp = re.sub(r"(?<=:)\d+",r"'\g<0>'",_temp)
+    # then properly convert to list of json obj
+    result = (_temp
+              .replace('\'','\"')
+              .replace('},','}')
+              .replace(' {','{')
+              .strip()
+              .split('\n'))
+    return result
+
+
+def find_ioc(hutch: str=None, patt: str=None):
+    """
+    A pythonic grep_ioc for gathering IOC details from the cfg file
+
+    Parameters
+    ----------
+    hutch : str, optional
+        3 letter lowercase hutch code. May also include 'all'.
+        The default is None.
+    patt : str, optional
+        Regex pattern to search for. The default is None.
+
+    Raises
+    ------
+    ValueError
+        Hutch code is invalid or regex pattern is missing.
+
+    Returns
+    -------
+    str
+        Returns the output as a list of dict after json formatting
+
+    """
+    hutch_list = ['xpp', 'xcs', 'cxi', 'mfx', 'mec', 'tmo', 'rix', 'xrt',
+		 'aux', 'det', 'fee', 'hpl', 'icl', 'las', 'lfe', 'kfe', 'tst',
+		 'thz', 'all']
```

Comment:
These sorts of lists may get hard to manage over time.

I suggest either adding the missing `txi` or pulling this from the subdirectories present in the /cds/group/pcds/pyps/config directory.

Reply:
Good catch! I think that's the safer option. I can just import that as a constant earlier in the script and refer to it and make it better to maintain.